File: Arrays & Hashing/4. Group Anagrams/solution.rb
# @param {String[]} strs
# @return {String[][]}
def group_anagrams(strs)
  map = {}
  strs.each do |str|
    key = str.chars.sort.join
    if map[key]
      map[key] = map[key].append(str)
    else
      map[key] = [str]
    end
  end
  return map.values
end

File: Arrays & Hashing/4. Group Anagrams/solution.exs
defmodule Solution do
  @spec group_anagrams(strs :: [String.t]) :: [[String.t]]
  def group_anagrams(strs) do
    strs
    |> Enum.map(&{&1 |> String.graphemes() |> Enum.sort(), &1})
    |> Enum.group_by(&elem(&1, 0), &elem(&1, 1))
    |> Enum.map(&elem(&1, 1))
  end
end
